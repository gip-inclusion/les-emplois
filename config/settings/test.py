import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
