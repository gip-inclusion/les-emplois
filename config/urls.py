from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
