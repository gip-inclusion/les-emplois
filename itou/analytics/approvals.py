from django.db.models import Count, Q

from itou.approvals import models as approvals_models
from itou.utils.apis import enums as api_enums

from . import models


def collect_approvals_counts(before):
    counts = approvals_models.Approval.objects.filter(created_at__lt=before).aggregate(
        total=Count("pk"),
        pe_notify_success=Count("pk", filter=Q(pe_notification_status=api_enums.PEApiNotificationStatus.SUCCESS)),
        pe_notify_pending=Count(
            "pk",
            filter=Q(
                pe_notification_status__in=(
                    api_enums.PEApiNotificationStatus.PENDING,
                    api_enums.PEApiNotificationStatus.SHOULD_RETRY,
                )
            ),
        ),
        pe_notify_error=Count("pk", filter=Q(pe_notification_status=api_enums.PEApiNotificationStatus.ERROR)),
    )
    return {
        models.DatumCode.APPROVAL_COUNT: counts["total"],
        models.DatumCode.APPROVAL_PE_NOTIFY_SUCCESS: counts["pe_notify_success"],
        models.DatumCode.APPROVAL_PE_NOTIFY_PENDING: counts["pe_notify_pending"],
        models.DatumCode.APPROVAL_PE_NOTIFY_ERROR: counts["pe_notify_error"],
    }


def collect_cancelled_approvals(before):
    cancelled_count = approvals_models.CancelledApproval.objects.filter(created_at__lt=before).count()
    return {models.DatumCode.APPROVAL_CANCELLED: cancelled_count}


def collect_analytics_data(before):
    return {
        **collect_approvals_counts(before),
        **collect_cancelled_approvals(before),
    }
