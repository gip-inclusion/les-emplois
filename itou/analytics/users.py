from django.db.models import Count

from itou.users import enums as users_enums, models as users_models

from . import models


def collect_analytics_data(before):
    counts = {
        info["kind"]: info["count"]
        for info in users_models.User.objects.values("kind").annotate(count=Count("pk")).values("kind", "count")
    }
    return {
        models.DatumCode.USER_COUNT: sum(counts.values()),
        models.DatumCode.USER_JOB_SEEKER_COUNT: counts.get(users_enums.KIND_JOB_SEEKER, 0),
        models.DatumCode.USER_PRESCRIBER_COUNT: counts.get(users_enums.KIND_PRESCRIBER, 0),
        models.DatumCode.USER_EMPLOYER_COUNT: counts.get(users_enums.KIND_EMPLOYER, 0),
        models.DatumCode.USER_LABOR_INSPECTOR_COUNT: counts.get(users_enums.KIND_LABOR_INSPECTOR, 0),
        models.DatumCode.USER_ITOU_STAFF_COUNT: counts.get(users_enums.KIND_ITOU_STAFF, 0),
    }
