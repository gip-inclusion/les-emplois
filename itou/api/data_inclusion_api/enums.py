import enum


class StructureTypeStr(str, enum.Enum):
    ORGA = "orga"
    SIAE = "siae"
