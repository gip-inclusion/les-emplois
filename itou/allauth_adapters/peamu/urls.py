from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
