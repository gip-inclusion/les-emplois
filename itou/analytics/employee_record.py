from django.db.models import Count, Max, Min

from itou.analytics import models
from itou.employee_record import models as employee_record_models


def collect_employee_records_count(before):
    return {
        models.DatumCode.EMPLOYEE_RECORD_COUNT: employee_record_models.EmployeeRecord.objects.filter(
            created_at__lt=before
        ).count(),
    }


def collect_probably_deleted_employee_records(before):
    data = employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before).aggregate(
        first_pk=Min("pk"),
        last_pk=Max("pk"),
        count_pk=Count("pk"),
    )

    value = ((data["last_pk"] - data["first_pk"] + 1) - data["count_pk"]) if None not in data.values() else 0
    return {models.DatumCode.EMPLOYEE_RECORD_DELETED: value}


def collect_employee_records_processing_code_of_first_exchange(before):
    data = (
        employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before)
        .values("asp_processing_code")
        .annotate(
            count=Count("asp_processing_code"),
        )
    )
    count_by_processing_code = {item["asp_processing_code"]: item["count"] for item in data}

    return {
        models.DatumCode.EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE: count_by_processing_code.get(
            employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE, 0
        ),
        models.DatumCode.EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE: sum(
            count
            for code, count in count_by_processing_code.items()
            if code != employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE
        ),
        models.DatumCode.EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE: count_by_processing_code.get(
            employee_record_models.EmployeeRecord.ASP_DUPLICATE_ERROR_CODE, 0
        ),
    }


def collect_employee_record_with_at_least_one_error(before):
    employee_records_with_at_least_one_error = employee_record_models.EmployeeRecordQuerySet.union(
        employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before)
        .exclude(
            asp_processing_code=employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE,
        )
        .values("pk"),
        employee_record_models.EmployeeRecordUpdateNotification.objects.filter(created_at__lt=before)
        .exclude(
            asp_processing_code=employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE,
        )
        .values("employee_record"),
    )

    value = (
        employee_record_models.EmployeeRecord.objects.filter(pk__in=employee_records_with_at_least_one_error)
        .distinct()
        .count()
    )
    return {models.DatumCode.EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR: value}


def collect_analytics_data(before):
    return {
        **collect_employee_records_count(before),
        **collect_probably_deleted_employee_records(before),
        **collect_employee_records_processing_code_of_first_exchange(before),
        **collect_employee_record_with_at_least_one_error(before),
    }
