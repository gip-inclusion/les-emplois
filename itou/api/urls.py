from django.urls import include, path
from drf_spectacular.views import SpectacularAPIView, SpectacularRedocView
from rest_framework import routers
from rest_framework.authtoken import views as auth_views

from itou.api.data_inclusion_api.views import DataInclusionStructureView

from .applicants_api.views import ApplicantsView
from .employee_record_api.viewsets import (
    DummyEmployeeRecordViewSet,
    EmployeeRecordUpdateNotificationViewSet,
    EmployeeRecordViewSet,
)
from .siae_api.viewsets import SiaeViewSet


# High level app for API
# https://docs.djangoproject.com/en/dev/topics/http/urls/#url-namespaces-and-included-urlconfs
app_name = "itou.api"

# Using DRF router with viewsets means automatic definition of utl patterns
router = routers.DefaultRouter()
router.register(r"employee-records", EmployeeRecordViewSet, basename="employee-records")
router.register(
    r"employee-record-notifications",
    EmployeeRecordUpdateNotificationViewSet,
    basename="employee-record-notifications",
)
router.register(r"dummy-employee-records", DummyEmployeeRecordViewSet, basename="dummy-employee-records")
router.register(r"siaes", SiaeViewSet, basename="siaes")

urlpatterns = [
    # TokenAuthentication endpoint to get token from login/password.
    path("token-auth/", auth_views.obtain_auth_token, name="token-auth"),
    # Needed for Browseable API (dev)
    path("api-auth/", include("rest_framework.urls", namespace="rest_framework")),
    # OpenAPI
    # See: https://www.django-rest-framework.org/topics/documenting-your-api/
    # OAS 3 YAML schema (downloadable)
    path(
        "oas3/",
        SpectacularAPIView.as_view(),
        name="openapi_schema",
    ),
    path(
        "redoc/",
        SpectacularRedocView.as_view(url_name="v1:openapi_schema"),
        name="redoc",
    ),
]

urlpatterns += router.urls

urlpatterns += [
    path("structures/", DataInclusionStructureView.as_view(), name="structures-list"),
    path("candidats/", ApplicantsView.as_view(), name="applicants-list"),
]
