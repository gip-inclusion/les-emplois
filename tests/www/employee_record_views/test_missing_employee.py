import datetime

from django.urls import reverse
from django.utils import timezone
from freezegun import freeze_time

from itou.job_applications.enums import JobApplicationState
from itou.www.employee_record_views.enums import MissingEmployeeCase
from tests.approvals.factories import ApprovalFactory
from tests.companies.factories import CompanyFactory
from tests.employee_record.factories import EmployeeRecordFactory
from tests.job_applications.factories import JobApplicationFactory
from tests.users.factories import JobSeekerFactory
from tests.utils.test import parse_response_to_soup


@freeze_time("2025-02-14")
def test_missing_employee(client, snapshot):
    siae = CompanyFactory(subject_to_eligibility=True, with_membership=True)
    user = siae.members.get()
    client.force_login(user)

    url = reverse("employee_record_views:missing_employee")

    response = client.get(url)
    assert str(parse_response_to_soup(response, selector=".s-section")) == snapshot(name="form")

    # job_seeker that was never hired
    job_seeker = JobSeekerFactory(first_name="André", last_name="Alonso")
    JobApplicationFactory(to_company=siae, job_seeker=job_seeker)
    response = client.post(url, data={"employee": job_seeker.pk})
    assert str(
        parse_response_to_soup(
            response,
            selector=".s-section",
            replace_in_attr=[
                (
                    "href",
                    f"/apply/siae/list?job_seeker={job_seeker.pk}",
                    "/apply/siae/list?job_seeker=[PK of Job seeker]",
                )
            ],
        )
    ) == snapshot(name=MissingEmployeeCase.NO_HIRING)

    # hired without approval (not possible anymore)
    job_seeker = JobSeekerFactory(first_name="Béatrice", last_name="Beauregard")
    JobApplicationFactory(to_company=siae, job_seeker=job_seeker, state=JobApplicationState.ACCEPTED, approval=None)
    response = client.post(url, data={"employee": job_seeker.pk})
    assert str(parse_response_to_soup(response, selector=".s-section")) == snapshot(
        name=MissingEmployeeCase.NO_APPROVAL
    )

    # hired in the future
    job_seeker = JobSeekerFactory(first_name="Charles", last_name="Constantin")
    approval = ApprovalFactory(user=job_seeker, number="XXXXXXX00001")
    JobApplicationFactory(
        to_company=siae,
        job_seeker=job_seeker,
        state=JobApplicationState.ACCEPTED,
        approval=approval,
        hiring_start_at=datetime.date(2025, 2, 15),
    )
    response = client.post(url, data={"employee": job_seeker.pk})
    assert str(parse_response_to_soup(response, selector=".s-section")) == snapshot(
        name=MissingEmployeeCase.FUTURE_HIRING
    )

    # But if there also was a previous accepted job application it works
    job_application = JobApplicationFactory(
        to_company=siae,
        job_seeker=job_seeker,
        state=JobApplicationState.ACCEPTED,
        approval=approval,
        hiring_start_at=datetime.date(2025, 1, 1),
        created_at=timezone.now() - datetime.timedelta(days=1),  # fallback for accepted_at
    )
    response = client.post(url, data={"employee": job_seeker.pk})
    assert response.context["approvals_data"][0][2] == MissingEmployeeCase.NO_EMPLOYEE_RECORD

    # hired and already has an employee record
    job_seeker = JobSeekerFactory(first_name="Damien", last_name="Danone")
    approval = ApprovalFactory(user=job_seeker, number="XXXXXXX00002")
    job_application = JobApplicationFactory(
        to_company=siae,
        job_seeker=job_seeker,
        state=JobApplicationState.ACCEPTED,
        approval=approval,
        hiring_start_at=datetime.date(2025, 2, 14),
    )
    employee_record = EmployeeRecordFactory(job_application=job_application)
    response = client.post(url, data={"employee": job_seeker.pk})
    assert str(
        parse_response_to_soup(
            response,
            selector=".s-section",
            replace_in_attr=[
                (
                    "href",
                    f"/employee_record/summary/{employee_record.pk}",
                    "/employee_record/summary/[PK of EmployeeRecord]",
                )
            ],
        )
    ) == snapshot(name=MissingEmployeeCase.EXISTING_EMPLOYEE_RECORD_SAME_COMPANY)

    # hired and already has an employee record in another siae on the same convention
    job_seeker = JobSeekerFactory(first_name="Eliott", last_name="Emery")
    approval = ApprovalFactory(user=job_seeker, number="XXXXXXX00003")
    JobApplicationFactory(
        to_company=siae,
        job_seeker=job_seeker,
        state=JobApplicationState.ACCEPTED,
        approval=approval,
        hiring_start_at=datetime.date(2025, 2, 14),
    )
    other_siae = CompanyFactory(convention=siae.convention, for_snapshot=True)
    job_application = JobApplicationFactory(
        to_company=other_siae,
        job_seeker=job_seeker,
        state=JobApplicationState.ACCEPTED,
        approval=approval,
        hiring_start_at=datetime.date(2025, 2, 14),
    )
    employee_record = EmployeeRecordFactory(job_application=job_application)
    response = client.post(url, data={"employee": job_seeker.pk})
    assert str(parse_response_to_soup(response, selector=".s-section")) == snapshot(
        name=MissingEmployeeCase.EXISTING_EMPLOYEE_RECORD_OTHER_COMPANY
    )

    # hired without an employee record
    job_seeker = JobSeekerFactory(first_name="Fabienne", last_name="Favriseau")
    approval = ApprovalFactory(user=job_seeker, number="XXXXXXX00004")
    job_application = JobApplicationFactory(
        to_company=siae,
        job_seeker=job_seeker,
        state=JobApplicationState.ACCEPTED,
        approval=approval,
        hiring_start_at=datetime.date(2025, 2, 14),
    )
    response = client.post(url, data={"employee": job_seeker.pk})
    assert str(
        parse_response_to_soup(
            response,
            selector=".s-section",
            replace_in_attr=[
                (
                    "href",
                    f"/employee_record/create/{job_application.pk}",
                    "/employee_record/create/[PK of JobApplication]",
                )
            ],
        )
    ) == snapshot(name=MissingEmployeeCase.NO_EMPLOYEE_RECORD)
