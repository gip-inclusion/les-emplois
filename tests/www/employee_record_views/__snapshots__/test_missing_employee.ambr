# serializer version: 1
# name: test_missing_employee[existing_employee_record_other_company]
  '''
  <section class="s-section">
      <div class="s-section__container container">
          <div class="row">
              <div class="col-12 col-md-9">
                  <div class="c-form mb-3 mb-lg-5">
                      <h2>
                          Eliott EMERY
                      </h2>
                      <p>
                          Ce salarié a 1 PASS IAE disponible
                      </p>
                      <hr/>
                      <p>
                          <strong>
                              XXXXXXX00003
                          </strong>
                          du 14/02/2025 au 13/02/2027
                      </p>
                      <p>
                          Une fiche salarié existe déjà pour ce PASS IAE dans une autre structure avec la même convention : EI Acme inc. - SIRET 012 345 678 910
                      </p>
                      <p>
                          Si vous souhaitez visualiser cette fiche salarié, vous devez vous connecter au tableau de bord de cette structure.
                      </p>
                      <p>
                          Seule la première embauche avec ce PASS IAE nécessite une transmission de la fiche salarié.
                          <br/>
                          En cas de besoin vous pouvez ajouter un nouveau contrat depuis votre espace Extranet IAE 2.0 de l’ASP.
                      </p>
                  </div>
              </div>
          </div>
      </div>
  </section>
  
  '''
# ---
# name: test_missing_employee[existing_employee_record_same_company]
  '''
  <section class="s-section">
      <div class="s-section__container container">
          <div class="row">
              <div class="col-12 col-md-9">
                  <div class="c-form mb-3 mb-lg-5">
                      <h2>
                          Damien DANONE
                      </h2>
                      <p>
                          Ce salarié a 1 PASS IAE disponible
                      </p>
                      <hr/>
                      <p>
                          <strong>
                              XXXXXXX00002
                          </strong>
                          du 14/02/2025 au 13/02/2027
                      </p>
                      <p>
                          Une fiche salarié existe déjà pour ce PASS IAE :
                          <a href="/employee_record/summary/[PK of EmployeeRecord]">
                              Voir la fiche salarié
                          </a>
                          .
                      </p>
                      <p>
                          Seule la première embauche avec ce PASS IAE, nécessite une transmission de la fiche salarié.
                          <br/>
                          En cas de besoin vous pouvez ajouter un nouveau contrat depuis votre espace Extranet IAE 2.0 de l’ASP.
                      </p>
                  </div>
              </div>
          </div>
      </div>
  </section>
  
  '''
# ---
# name: test_missing_employee[form]
  '''
  <section class="s-section">
      <div class="s-section__container container">
          <div class="row">
              <div class="col-12 col-md-9">
                  <div class="c-form mb-3 mb-lg-5">
                      <h2>
                          Rechercher parmi tous les salariés
                      </h2>
                      <form method="post">
                          <input name="csrfmiddlewaretoken" type="hidden" value="NORMALIZED_CSRF_TOKEN"/>
                          <p>
                              Dans le menu déroulant ci-dessous, vous trouverez la liste de vos salariés et candidats pour lesquels il y a au moins une candidature au sein de votre structure.
                          </p>
                          <div class="form-group form-group-required">
                              <label class="form-label" for="id_employee">
                                  Nom du salarié
                              </label>
                              <select class="form-select django-select2" data-allow-clear="false" data-minimum-input-length="0" data-theme="bootstrap-5" id="id_employee" lang="fr" name="employee" required="">
                                  <option selected="" value="">
                                      Sélectionnez le salarié
                                  </option>
                              </select>
                          </div>
                          <div class="row">
                              <div class="col-12">
                                  <hr class="mb-3"/>
                                  <small class="d-inline-block mb-3">
                                      * champs obligatoires
                                  </small>
                                  <div class="form-row align-items-center justify-content-end gx-3">
                                      <div class="form-group col-12 col-lg order-3 order-lg-1">
                                          <a aria-label="Annuler la saisie de ce formulaire" class="btn btn-link btn-ico ps-lg-0 w-100 w-lg-auto" href="/employee_record/list">
                                              <i aria-hidden="true" class="ri-close-line ri-lg">
                                              </i>
                                              <span>
                                                  Annuler
                                              </span>
                                          </a>
                                      </div>
                                      <div class="form-group col col-lg-auto order-2 order-lg-3">
                                          <button aria-label="Passer à l’étape suivante" class="btn btn-block btn-primary" type="submit">
                                              <span>
                                                  Rechercher
                                              </span>
                                          </button>
                                      </div>
                                  </div>
                              </div>
                          </div>
                      </form>
                  </div>
              </div>
          </div>
      </div>
  </section>
  
  '''
# ---
# name: test_missing_employee[no_approval]
  '''
  <section class="s-section">
      <div class="s-section__container container">
          <div class="row">
              <div class="col-12 col-md-9">
                  <div class="c-form mb-3 mb-lg-5">
                      <h2>
                          Béatrice BEAUREGARD
                      </h2>
                      <p>
                          La création d’une fiche salarié pour cette personne est impossible car vous aviez déclaré une embauche sans aide au poste. Par conséquent, elle ne possède pas de PASS IAE.
                      </p>
                      <p>
                          Si vous souhaitez bénéficier d’une aide au poste, vous devez d’abord obtenir un PASS IAE en effectuant une nouvelle embauche, puis créer sa fiche salarié.
                      </p>
                  </div>
              </div>
          </div>
      </div>
  </section>
  
  '''
# ---
# name: test_missing_employee[no_employee_record]
  '''
  <section class="s-section">
      <div class="s-section__container container">
          <div class="row">
              <div class="col-12 col-md-9">
                  <div class="c-form mb-3 mb-lg-5">
                      <h2>
                          Fabienne FAVRISEAU
                      </h2>
                      <p>
                          Ce salarié a 1 PASS IAE disponible
                      </p>
                      <hr/>
                      <p>
                          <strong>
                              XXXXXXX00004
                          </strong>
                          du 14/02/2025 au 13/02/2027
                      </p>
                      <a href="/employee_record/create/[PK of JobApplication]">
                          Créer la fiche salarié
                      </a>
                  </div>
              </div>
          </div>
      </div>
  </section>
  
  '''
# ---
# name: test_missing_employee[no_hiring]
  '''
  <section class="s-section">
      <div class="s-section__container container">
          <div class="row">
              <div class="col-12 col-md-9">
                  <div class="c-form mb-3 mb-lg-5">
                      <h2>
                          André ALONSO
                      </h2>
                      <p>
                          Vous n’avez pas déclaré d’embauche pour ce candidat. Pour créer une fiche salarié vous devez accepter la candidature.
                          <br/>
                          Vous pouvez trouver la liste des candidatures de ce candidat en
                          <a href="/apply/siae/list?job_seeker=[PK of Job seeker]&archived=all">
                              cliquant ici
                          </a>
                          .
                      </p>
                  </div>
              </div>
          </div>
      </div>
  </section>
  
  '''
# ---
